app/assets/javascripts/App.js

```diff
@@ -9,10 +9,11 @@ import Header from './components/Header';
 import Footer from './components/Footer';
 import Home from './pages/Home';
 
-const App = React.createClass({
-
-  mixins: [Router.State],
+class App {
```

Comment:
this would become `export default class App` if the `mixin` approach you are proposing works.